### PR TITLE
Add folly dependency

### PR DESCRIPTION
**Pull request by bertmaher**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,27 @@ else()
...
+    set(FOLLY_HAVE_WEAK_SYMBOLS ON CACHE BOOL "Compiler supports weak symbols")
+    set(OPENSSL_ROOT_DIR /usr/local/opt/openssl)
+    find_package(OpenSSL REQUIRED)
+  endif()
+
+  # Remove Glow-specific CMAKE_CXX_FLAGS to build folly.
+  set(SAVE_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  set(CMAKE_CXX_FLAGS "")
+  add_subdirectory("${GLOW_THIRDPARTY_DIR}/folly" EXCLUDE_FROM_ALL)
+  target_include_directories(folly PUBLIC "${OPENSSL_INCLUDE_DIR}")
```
Comment (jackm321) [inline CMakeLists.txt]: OPENSSL_INCLUDE_DIR?

Comment (jackm321) [reply CMakeLists.txt]: Oh nvm, needed by folly I guess

Comment (bertmaher) [reply CMakeLists.txt]: Yeah I dunno why folly’s internal build system doesn’t figure this out properly but there you have it.

**Comment by bertmaher** [comment]
@ayermolo Here's my second take at importing folly.  This PR only requires boost, double-conversion, and jemalloc as external dependencies.  I've taken a *very* aggressive subset of folly.  Possibly *too* aggressive, since people who want to use folly will need to wade through a few pages of missing symbol errors until they link all the sources they want.  But, it ensures we don't pull in too much stuff.

**Comment by ayermolo** [comment]
@bertmaher Really appreciate it. I am trying to wrap up something internally, will try to pull this branch in by EOW to try to build on windows.

**Comment by bertmaher** [comment]
Unfortunately this approach is not working on Linux.  I'm going to revert to the original approach.

**Comment by ayermolo** [comment]
That is unfortunate. What was the original approach bring in all the dependencies?

**Comment by bertmaher** [comment]
Yeah, bringing in the deps.  Tbh, the only one I was able to avoid was openssl with the other approach.

**Comment by ayermolo** [comment]
I see. Thank you for trying.

**Pull request closed without merging**